flow/model/qvalue/qvalue.go

```diff
@@ -21,157 +21,603 @@ import (
+func (v QValueCIDR) Value() any {
+	return v.Val
+}
+
+type QValueINET struct {
+	Val string
+}
+
+func (QValueINET) Kind() QValueKind {
+	return QValueKindINET
+}
+
+func (v QValueINET) Value() any {
+	return v.Val
+}
+
+type QValueMacaddr struct {
+	Val string
+}
+
+func (QValueMacaddr) Kind() QValueKind {
+	return QValueKindMacaddr
+}
+
+func (v QValueMacaddr) Value() any {
+	return v.Val
+}
+
+type QValueArrayFloat32 struct {
+	Val []float32
+}
+
+func (QValueArrayFloat32) Kind() QValueKind {
+	return QValueKindArrayFloat32
+}
+
+func (v QValueArrayFloat32) Value() any {
+	return v.Val
+}
+
+type QValueArrayFloat64 struct {
+	Val []float64
+}
+
+func (QValueArrayFloat64) Kind() QValueKind {
+	return QValueKindArrayFloat64
+}
+
+func (v QValueArrayFloat64) Value() any {
+	return v.Val
+}
+
+type QValueArrayInt16 struct {
+	Val []int16
+}
+
+func (QValueArrayInt16) Kind() QValueKind {
+	return QValueKindInt16
+}
+
+func (v QValueArrayInt16) Value() any {
+	return v.Val
+}
+
+type QValueArrayInt32 struct {
+	Val []int32
+}
+
+func (QValueArrayInt32) Kind() QValueKind {
+	return QValueKindInt32
+}
+
+func (v QValueArrayInt32) Value() any {
+	return v.Val
+}
+
+type QValueArrayInt64 struct {
+	Val []int64
+}
+
+func (QValueArrayInt64) Kind() QValueKind {
+	return QValueKindArrayInt64
+}
+
+func (v QValueArrayInt64) Value() any {
+	return v.Val
+}
+
+type QValueArrayString struct {
+	Val []string
+}
+
+func (QValueArrayString) Kind() QValueKind {
+	return QValueKindArrayString
+}
+
+func (v QValueArrayString) Value() any {
+	return v.Val
+}
+
+type QValueArrayDate struct {
+	Val []time.Time
+}
+
+func (QValueArrayDate) Kind() QValueKind {
+	return QValueKindArrayDate
+}
+
+func (v QValueArrayDate) Value() any {
+	return v.Val
+}
+
+type QValueArrayTimestamp struct {
+	Val []time.Time
+}
+
+func (QValueArrayTimestamp) Kind() QValueKind {
+	return QValueKindArrayTimestamp
+}
+
+func (v QValueArrayTimestamp) Value() any {
+	return v.Val
+}
+
+type QValueArrayTimestampTZ struct {
+	Val []time.Time
+}
+
+func (QValueArrayTimestampTZ) Kind() QValueKind {
+	return QValueKindArrayTimestampTZ
+}
+
+func (v QValueArrayTimestampTZ) Value() any {
+	return v.Val
+}
+
+type QValueArrayBoolean struct {
+	Val []bool
+}
+
+func (QValueArrayBoolean) Kind() QValueKind {
+	return QValueKindArrayBoolean
+}
+
+func (v QValueArrayBoolean) Value() any {
+	return v.Val
+}
+
+func Equals(qv QValue, other QValue) bool {
+	qvValue := qv.Value()
+	otherValue := other.Value()
+	qvEmpty := qvValue == nil || qvValue == "" || qvValue == "null" ||
+		(reflect.TypeOf(qvValue).Kind() == reflect.Slice && reflect.ValueOf(qvValue).Len() == 0)
+	otherEmpty := otherValue == nil || otherValue == "" || otherValue == "null" ||
+		(reflect.TypeOf(otherValue).Kind() == reflect.Slice && reflect.ValueOf(otherValue).Len() == 0)
+	if qvEmpty && otherEmpty {
+		return true
+	}
+	qvString, ok1 := qvValue.(string)
+	otherString, ok2 := otherValue.(string)
+	if ok1 && ok2 && qvString == otherString {
```

Comment:
why special case string and not just use the switch?

Reply:
Because our tests rely on a multiple cases of `qvalue.Equals(QValueINET{Val:"aa"}, QValueString{Val:"aa"})` being true